tc-server/src/main/java/com/tc/objectserver/entity/ManagedEntityImpl.java

```diff
@@ -1006,10 +1018,18 @@ private void getEntity(ServerEntityRequest getEntityRequest, ResultCapture respo
               handler.handleReconnect(descriptor, extendedData);
             }
           } catch (ReconnectRejectedException rejected) {
+            Assert.assertTrue(clientEntityStateManager.removeReference(descriptor));
```

Comment:
Should the failure of this assertion (and others) be tunnelled to the client side too?

Reply:
This is a true assertion because the add happened inline.  The rejection is tunneled back to the client.